src/Controller/UploadController.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\dropzonejs\Controller\UploadController.
+ */
+
+namespace Drupal\dropzonejs\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\dropzonejs\UploadException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * Handles requests that dropzone issues when uploading files.
+ */
+class UploadController extends ControllerBase {
+
+  /**
+   * The current request.
+   *
+   * @var \Symfony\Component\HttpFoundation\Request $request
+   *   The HTTP request object.
+   */
+  protected $request;
+
+  /**
+   * Stores temporary folder URI.
+   *
+   * This is configurable via the configuration variable. It was added for HA
+   * environments where temporary location may need to be a shared across all
+   * servers.
+   *
+   * @var string
+   */
+  protected $temporaryUploadLocation;
+
+  /**
+   * Filename of a file that is being uploaded.
+   *
+   * @var string
+   */
+  protected $filename;
+
+  /**
+   * Constructs dropzone upload controller route controller.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   Request object.
+   */
+  public function __construct(Request $request) {
+    $this->request = $request;
+    $this->temporaryUploadLocation = \Drupal::config('system.file')->get('path.temporary');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static($container->get('request_stack')->getCurrentRequest());
+  }
+
+  /**
+   * Handles DropzoneJs uploads.
+   */
+  public function handleUploads() {
+    // @todo: Implement file_validate_size();
+    try {
+      $this->prepareTemporaryUploadDestination();
+      $this->handleUpload();
+    }
+    catch (UploadException $e) {
+      return $e->getErrorResponse();
+    }
+
+    // Return JSON-RPC response.
+    // @todo Not sure if we need this.
+    return new JsonResponse(
```

Comment:
Is this from Plupload? Then we probably don't need it or must be adapted to dropzone.